src/core/settings/qgssettingsentry.cpp

```diff
@@ -83,6 +83,58 @@ QString QgsSettingsEntryBase::key( const QStringList &dynamicKeyPartList ) const
   return completeKey;
 }
 
+bool QgsSettingsEntryBase::checkKey( const QString &key ) const
+{
+  // Key to check
+  QString completeKeyToCheck = key;
+
+  QString settingsPrefix = QgsSettings().prefixedKey( "", section() );
```

Comment:
```suggestion
  QString settingsPrefix = QgsSettings().prefixedKey( QString(), section() );
```

I think you are also missing `QStringLiteral` and/or a char literal instead of (as an example) `"/"` in the rest of the code.

Reply:
Also fixed inclusive missing QStringLiterals